tests/mapped_struct.py

```diff
@@ -2082,10 +2122,24 @@ def testProxiedTupleRepr(self):
         c = self.pack([1, 2.0])
         self.assertEquals(repr(c), "proxied_tuple((1,2.0))")
 
+def tuple_cmp(a, b):
+    for ea, eb in zip(a,b):
+        if key_cmp(ea, eb):
+            return key_cmp(ea, eb)
+    return 0
+
+def key_cmp(a, b):
+    if type(a) != type(b):
+        return cmp(type(a).__name__, type(b).__name__)
+    if isinstance(a, tuple):
+        return tuple_cmp(a, b)
+    return cmp(a, b)
 
 class DictPackingCommonTest(object):
 
     def assertUnsortedEquals(self, a, b):
+        if six.PY3:
+            return sorted(a, key=functools.cmp_to_key(key_cmp)) == sorted(b, key=functools.cmp_to_key(key_cmp))
```

Comment:
I don't follow why you need this. Sorting item tuples should work fine in both python versions.

Reply:
No, it's comparing int and str which doesn't work in py3. For example, from one of the values in MappedDictPackingTest:

in py2
```
sorted(['a', 1, frozenset(), (1, 2)]) == [1, frozenset([]), 'a', (1, 2)]
```
in py3
```
>>> sorted(['a', 1, frozenset(), (1, 2)])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'int' and 'str'
```

Reply:
Oh... it's a mixed-type test, didn't notice.

What about `key=repr`?